api/lib/opentelemetry/metrics/time_series.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    # TimeSeries is a base class for a time series.
+    class TimeSeries
+      def add(value); end
+
+      def set(value); end
+
+      # @api private
+      module Double
+        def add(value)
+          raise ArgumentError if value.integer?
```

Comment:
I think we should drop this check. It should be legal to pass an integer, as long as it is within the double-precision floating point range.